### ISSUE TITLE
Error building with latest graphcache and cra

### ISSUE THREAD

**Issue by zenios** [FormidableLabs/urql#708]
Hi I get the following error when building cra app with the latest graphcache

```
./node_modules/@urql/exchange-populate/dist/urql-exchange-populate.mjs
Attempted import error: 'visitWithTypeInfo' is not exported from 'graphql/utilities/TypeInfo'.
```

Thanks

**Comment by coding-days** [comment]
hello, I still get this error :

```
[ error ] /node_modules/@urql/exchange-populate/dist/urql-exchange-populate.mjs
Attempted import error: 'visitWithTypeInfo' is not exported from 'graphql/utilities/TypeInfo'.
```

While I have this in my package.json, full re-install :

```
"next-urql": "0.3.7",
"urql": "1.9.7",
"@urql/devtools": "1.0.2",
"@urql/exchange-graphcache"
```

I tried to lower versions and all, still having the same problem :/
thank you for your help !

**Comment by JoviDeCroock** [comment]
Do you by any chance have an ejected webpack config?

**Comment by coding-days** [comment]
I just have a simple next.config.js from Next.js :

```
module.exports = withPlugins(
  [
    withTM({
      experimental: {
        scss: true,
      },
    }),
    withBundleAnalyzer,
    withImages,
  ],
  {
    target: 'serverless',
    env: envVars,
    webpack: (config, { isServer }) => {
      if (isDev) {
        config.plugins = [
          ...config.plugins,
          new Dotenv({
            path: path.join(__dirname, '.env.build'),
            systemvars: true,
          }),
        ];
      }
      if (!isServer) {
        config.node = {
          fs: 'empty',
          net: 'empty',
          tls: 'empty',
          dns: 'empty',
        };
      }

      config.plugins.push(new LodashModuleReplacementPlugin({}));

      config.module.rules.push({
        use: {
          loader: 'next-babel-loader',
          options: {
            isServer,
            cwd: process.cwd(),
            plugins: [['lodash', { id: ['lodash-es'] }]],
          },
        },
        test: /\.js$/,
        exclude: /node_modules/,
        include: [process.cwd()],
      });

      const originalEntry = config.entry;
      config.entry = async () => {
        const entries = await originalEntry();
        if (
          entries['main.js'] &&
          !entries['main.js'].includes('./lib/polyfills.ts')
        ) {
          entries['main.js'].unshift('./lib/polyfills.ts');
        }
        return entries;
      };

      return config;
    },
  },
);
```

Which didn't really change neither ^^

**Comment by JoviDeCroock** [comment]
This is about CRA though, the issue you talk about could be caused by a multitude of factors. The thing that comes to mind is the breaking change `graphql` made in their transition from 14 --> 15

**Comment by kitten** [comment]
> I tried to lower versions and all, still having the same problem :/

@coding-days: Have you upgraded your packages though, specifically the populate exchange? This was an issue that simply affected some specific versions, so it’s frankly impossible to occur once you’re off those versions 😅

**Comment by coding-days** [comment]
I forced the `"graphql": "14.6.0"` but still having the issue.
@kitten: I never installed the populate exchange, I'll give it a try !

**Comment by coding-days** [comment]
So I have :

```
"graphql": "14.6.0",
"next-urql": "0.3.7",
"urql": "1.9.7",
"@urql/devtools": "1.0.2",
"@urql/exchange-graphcache": "2.4.0",
"@urql/exchange-populate": "^0.1.6",
```

When I check in the node_modules `graphql/utilities/TypeInfo`, I don't see any `visitWithTypeInfo` exported indeed. Must I use the `graphql@15` ?

**Comment by kitten** [comment]
@coding-days the populate exchange itself is currently a deprecated dependency if Graphcache.

So if you run `yarn upgrade @urql/exchange-populate` it should fix this issue

**Comment by coding-days** [comment]
Oh okay, I'll kinda relieved because `yarn upgrade @urql/exchange-populate` didn't work either, did a full re-install again with `yarn.lock` deleted and I was starting to feel really confused :p

**Comment by kitten** [comment]
@coding-days This should be fixed with `@urql/exchange-populate@0.1.7`
